src/config/mod.rs

```diff
@@ -32,13 +29,27 @@ impl Config {
     ) -> Option<Self> {
         // get directory
         let mut config_dir_path = if let Some(config_dir) = config_directory {
-            let config_dir_path = PathBuf::from(config_dir);
+            if let Some(_profile) = profile {
+                let config_dir_path = PathBuf::from(config_dir);
 
-            if config_dir_path.exists() {
-                // use this config
-                config_dir_path
+                if config_dir_path.exists() {
+                    // use this config
+                    config_dir_path
+                } else {
+                    panic!("The provided config directory is invalid");
+                }
             } else {
-                panic!("The provided config directory is invalid");
+                let mut error = StyledStr::new();
+
+                error.push_str(Some(Style::Error), "error: ".to_string());
+                error.push_str(
```

Comment:
Do we need to build our own error here? I was under the impression clap lets you specify the one arg is only required when another is passed.